### PR TITLE
Cart page with related components and entity, and currency formatter

### PR DESCRIPTION
**Pull request by Eunji-Elly-Lee**
**Jira Ticket**<br>
Link: [[Frontend] Cart Page UI](https://derek-team-lva74agprdbm.atlassian.net/browse/SP-48?atlOrigin=eyJpIjoiMjE3YmE2ODA2MWVhNDFlYjk1NDA4MGFlODMzNTM0MzIiLCJwIjoiaiJ9)

**Description**<br>
_Add a sensible description to this pull request:_
Update cart page with cart-related components (Cart, CartItemCard, CartSummary) and entity(CartItem), and currency formatter utility

- Added: `components/carts/Cart.tsx` `components/carts/CartItemCard.tsx` `components/carts/CartSummary.tsx` `entities/CartItem.ts` `utils/formatter.ts`
- Changed: `app/(home)/cart/page.tsx`
- Refactored:
- Removed:

**Screenshots**<br>
Screenshot 2024-04-20 230858

### CODE REVIEW

**Comment by Euna-kim-1** [comment]
I think we don't need comments everywhere, just in the really crucial parts..? so we could remove the comments that state the obvious.

**Comment by JenHLee** [comment]
Screenshot 2024-04-28 at 11 19 02 PM

As far as I know, it seems like there was an agreement to capitalize only the first letter of button text for design consistency. If you check in Figma, you should be able to see that it's "Checkout" rather than "CHECKOUT."

**Comment by Eunji-Elly-Lee** [comment]
> Screenshot 2024-04-28 at 11 19 02 PM
> As far as I know, it seems like there was an agreement to capitalize only the first letter of button text for design consistency. If you check in Figma, you should be able to see that it's "Checkout" rather than "CHECKOUT."

image
As shown in the attached screenshot, I designed the text on the checkout button to be all capitalized, referencing the Component section of Figma. However, I should have confirmed that the text style of the checkout button was consistent across all sections. Based on your feedback, I've corrected the text style of the checkout button to match the design guidelines, ensuring that the first letter is capitalized and the rest are lowercase. Thank you!

File: components/carts/CartItemCard.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  updateQuantity: (cartItem: CartItem, newQuantity: number) => void;
+  removeCartItem: (cartItem: CartItem) => void;
+}
+
+const CartItemCard: React.FC<CartItemProps> = ({
+  cartItem,
+  updateQuantity,
+  removeCartItem,
+}) => {
+  // Decrease the quantity of the cart item by 1 (if quantity is greater than 1)
```
Comment (HarryLee87) [inline components/carts/CartItemCard.tsx]: I do not think this comment is necessary.

**Comment by sj0020** [comment]
Harry and I reviewed it together and it looks good to us!

**Review by HarryLee87 — APPROVED** [review]

**Review by HarryLee87 — APPROVED** [review]

**Review by HarryLee87 — APPROVED** [review]

**Review by HarryLee87 — APPROVED** [review]

**Pull request merged**